addons/compute_worker/gpu_uniforms/gpu_uniform.gd

```diff
@@ -138,22 +138,67 @@ func byte_array_to_vec3(array: PackedByteArray) -> Vector3:
 	return vec
 
 
-## Convert a Vector3 to GLSL equivalent `dvec3, dvec4` format
-func vec3_to_byte_array(vector: Vector3) -> PackedByteArray:
+## Convert a Vector3 to GLSL equivalent `dvec3` format
+func dvec3_to_byte_array(vector: Vector3) -> PackedByteArray:
 	
 	return PackedFloat64Array([vector.x, vector.y, vector.z, 0.0]).to_byte_array()
 
+## Convert a Vector3 to GLSL equivalent `vec3` format
+func vec3_to_byte_array(vector: Vector3) -> PackedByteArray:
+	
+	return PackedFloat32Array([vector.x, vector.y, vector.z, 0.0]).to_byte_array()
 
-## Convert a Vector3i to GLSL equivalent `ivec3, ivec4` format
+
+## Convert a Vector3i to GLSL equivalent `ivec3` format
 func vec3i_to_byte_array(vector: Vector3i) -> PackedByteArray:
 	
 	# We have to add a value for the "w" field for the vector,
 	# because the alignment spec for GLSL vec3s requires 16bytes
 	return PackedInt32Array([vector.x, vector.y, vector.z, 0]).to_byte_array()
 
+func vec2i_to_byte_array(vector: Vector2i) -> PackedByteArray:
+	
+	# We have to add a value for the "z" field for the vector,
+	# because the alignment spec for GLSL vec2s requires 16bytes
+	return PackedInt32Array([vector.x, vector.y, 0]).to_byte_array()
+
 
-## Convert an array of Vector3s to GLSL equivalent `vec3[], vec4[]` format
-func vec3_array_to_byte_array(array: PackedVector3Array):
+## Convert an array of Vector3s to GLSL equivalent `vec2[]` format
+func vec2_array_to_byte_array(array: PackedVector2Array):
+	
+	var bytes: PackedByteArray = PackedByteArray()
+	
+	for vector in array:
+		var vec: Vector3 = Vector3()
+		
+		vec.x = vector.x
+		vec.y = vector.y
+
+		
+		
+		var float_arr = PackedFloat32Array([vec.x, vec.y]).to_byte_array()
+		bytes.append_array(float_arr)
```

Comment:
Vectors that are inside arrays in GLSL have their alignment rounded up to a multiple of 4 of the base alignment of its elements. So here, where we define `float_arr`, we would add two more components to the array to fill out the rest of the array element's space inside the array.